verify.js

```diff
@@ -165,15 +165,13 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
   }
 
   if (options.maxAge) {
-    var maxAge = ms(options.maxAge);
     if (typeof payload.iat !== 'number') {
       return done(new JsonWebTokenError('iat required when maxAge is specified'));
     }
-    // We have to compare against either options.clockTimestamp or the currentDate _with_ millis
-    // to not change behaviour (version 7.2.1). Should be resolve somehow for next major.
-    var nowOrClockTimestamp = ((options.clockTimestamp || 0) * 1000) || Date.now();
-    if (nowOrClockTimestamp - (payload.iat * 1000) > maxAge + (options.clockTolerance || 0) * 1000) {
-      return done(new TokenExpiredError('maxAge exceeded', new Date(payload.iat * 1000 + maxAge)));
+
+    var maxAgeTimestamp = timespan(options.maxAge, payload.iat);
```

Comment:
Are you aware that your if `options.maxAge` was created by a `new String(...)` call, this will always return an error? Your `timespan` method does a `typeof x === 'string'` but `typeof new String('foo') === object`.

Reply:
Ok thanks, in that case ms(ms(ms())) might also do the trick...

Reply:
Ok, thanks, got it.

Reply:
To add to the conversation above, both `ms(String('1s'))` and `ms(String(1000))` work. No need to use new, as [String](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) object can be used as a global constructor for strings.

Reply:
Well, it is the consumer who is deciding what to send. But thanks to your comment I realized I didn't add a test for `String('12s')`, I'll add it later.

Reply:
Makes sense - thanks for noting that!